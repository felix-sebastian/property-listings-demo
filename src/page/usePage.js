import { useState, useEffect } from "react";
import axios from "axios";

const propertiesUrl =
  "https://cors-anywhere.herokuapp.com/https://code-challenge.activepipe.com/challenge/properties";

export default () => {
  const [state, setState] = useState("LOADING");
  const [status, setStatus] = useState(null);
  const [properties, setProperties] = useState([]);

  useEffect(() => {
    axios
      .get(propertiesUrl)
      .then(result => {
        if (result.status !== 200) setState("ERROR");
        else {
          setState("OK");
          setProperties(result.data);
        }
      })
      .catch(() => {
        setState("ERROR");
      });
  }, []);

  return {
    state,
    properties: status
      ? properties.filter(property => property.status === status)
      : properties,
    status,
    setStatus
  };
};
