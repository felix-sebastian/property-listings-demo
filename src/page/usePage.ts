import { useState, useEffect } from "react";
import axios from "axios";

const propertiesUrl =
  "https://cors-anywhere.herokuapp.com/https://code-challenge.activepipe.com/challenge/properties";

export default (): {
  state: string;
  properties: any;
  status: string | null;
  setStatus: Function;
} => {
  const [state, setState] = useState<string>("LOADING");
  const [status, setStatus] = useState<string | null>(null);
  const [properties, setProperties] = useState<any>([]);

  useEffect(() => {
    (async () => {
      try {
        const result = await axios.get(propertiesUrl);

        if (result.status !== 200) setState("ERROR");
        else {
          setState("OK");
          setProperties(result.data);
        }
      } catch {
        setState("ERROR");
      }
    })();
  }, []);

  return {
    state,
    properties: status
      ? properties.filter(
          (property: { status: string }) => property.status === status
        )
      : properties,
    status,
    setStatus
  };
};
