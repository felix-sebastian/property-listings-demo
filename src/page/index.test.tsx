import React from "react";
import { render } from "@testing-library/react";
import Page from "./";
import usePage from "./usePage";

jest.mock("./usePage", () =>
  jest.fn().mockReturnValue({
    state: "LOADING",
    properties: [],
    status: null,
    setStatus: jest.fn()
  })
);

test("renders loading state", () => {
  expect(render(<Page />)).toMatchSnapshot();
});

test("renders ok state", () => {
  (usePage as jest.Mock).mockReturnValueOnce({
    state: "OK",
    properties: [{ id: 1, foo: "bar" }],
    status: null,
    setStatus: jest.fn()
  });

  expect(render(<Page />)).toMatchSnapshot();
});

test("renders error state", () => {
  (usePage as jest.Mock).mockReturnValueOnce({
    state: "ERROR",
    properties: [],
    status: null,
    setStatus: jest.fn()
  });

  expect(render(<Page />)).toMatchSnapshot();
});
