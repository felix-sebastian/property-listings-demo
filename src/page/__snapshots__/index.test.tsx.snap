// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders error state 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fznZeY jliwHV"
      >
        <div
          class="sc-fzplWN gVyXxh"
        >
          <div
            class="sc-fznyAO ffbWNH"
          >
            <div>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-exclamation-circle fa-w-16 "
                data-icon="exclamation-circle"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                  fill="currentColor"
                />
              </svg>
               Uh oh... it looks like some things haven't loaded correctly
            </div>
            <div>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-sync fa-w-16 sc-fznKkj kWWQAh"
                data-icon="sync"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M440.65 12.57l4 82.77A247.16 247.16 0 0 0 255.83 8C134.73 8 33.91 94.92 12.29 209.82A12 12 0 0 0 24.09 224h49.05a12 12 0 0 0 11.67-9.26 175.91 175.91 0 0 1 317-56.94l-101.46-4.86a12 12 0 0 0-12.57 12v47.41a12 12 0 0 0 12 12H500a12 12 0 0 0 12-12V12a12 12 0 0 0-12-12h-47.37a12 12 0 0 0-11.98 12.57zM255.83 432a175.61 175.61 0 0 1-146-77.8l101.8 4.87a12 12 0 0 0 12.57-12v-47.4a12 12 0 0 0-12-12H12a12 12 0 0 0-12 12V500a12 12 0 0 0 12 12h47.35a12 12 0 0 0 12-12.6l-4.15-82.57A247.17 247.17 0 0 0 255.83 504c121.11 0 221.93-86.92 243.55-201.82a12 12 0 0 0-11.8-14.18h-49.05a12 12 0 0 0-11.67 9.26A175.86 175.86 0 0 1 255.83 432z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fznZeY jliwHV"
    >
      <div
        class="sc-fzplWN gVyXxh"
      >
        <div
          class="sc-fznyAO ffbWNH"
        >
          <div>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-exclamation-circle fa-w-16 "
              data-icon="exclamation-circle"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
                fill="currentColor"
              />
            </svg>
             Uh oh... it looks like some things haven't loaded correctly
          </div>
          <div>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-sync fa-w-16 sc-fznKkj kWWQAh"
              data-icon="sync"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M440.65 12.57l4 82.77A247.16 247.16 0 0 0 255.83 8C134.73 8 33.91 94.92 12.29 209.82A12 12 0 0 0 24.09 224h49.05a12 12 0 0 0 11.67-9.26 175.91 175.91 0 0 1 317-56.94l-101.46-4.86a12 12 0 0 0-12.57 12v47.41a12 12 0 0 0 12 12H500a12 12 0 0 0 12-12V12a12 12 0 0 0-12-12h-47.37a12 12 0 0 0-11.98 12.57zM255.83 432a175.61 175.61 0 0 1-146-77.8l101.8 4.87a12 12 0 0 0 12.57-12v-47.4a12 12 0 0 0-12-12H12a12 12 0 0 0-12 12V500a12 12 0 0 0 12 12h47.35a12 12 0 0 0 12-12.6l-4.15-82.57A247.17 247.17 0 0 0 255.83 504c121.11 0 221.93-86.92 243.55-201.82a12 12 0 0 0-11.8-14.18h-49.05a12 12 0 0 0-11.67 9.26A175.86 175.86 0 0 1 255.83 432z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders loading state 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fznZeY jliwHV"
      >
        <div
          class="sc-fzoLsD tlPGD"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-notch fa-w-16 sc-fzpans ljoJNU"
            data-icon="circle-notch"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M288 39.056v16.659c0 10.804 7.281 20.159 17.686 23.066C383.204 100.434 440 171.518 440 256c0 101.689-82.295 184-184 184-101.689 0-184-82.295-184-184 0-84.47 56.786-155.564 134.312-177.219C216.719 75.874 224 66.517 224 55.712V39.064c0-15.709-14.834-27.153-30.046-23.234C86.603 43.482 7.394 141.206 8.003 257.332c.72 137.052 111.477 246.956 248.531 246.667C393.255 503.711 504 392.788 504 256c0-115.633-79.14-212.779-186.211-240.236C302.678 11.889 288 23.456 288 39.056z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fznZeY jliwHV"
    >
      <div
        class="sc-fzoLsD tlPGD"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-notch fa-w-16 sc-fzpans ljoJNU"
          data-icon="circle-notch"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M288 39.056v16.659c0 10.804 7.281 20.159 17.686 23.066C383.204 100.434 440 171.518 440 256c0 101.689-82.295 184-184 184-101.689 0-184-82.295-184-184 0-84.47 56.786-155.564 134.312-177.219C216.719 75.874 224 66.517 224 55.712V39.064c0-15.709-14.834-27.153-30.046-23.234C86.603 43.482 7.394 141.206 8.003 257.332c.72 137.052 111.477 246.956 248.531 246.667C393.255 503.711 504 392.788 504 256c0-115.633-79.14-212.779-186.211-240.236C302.678 11.889 288 23.456 288 39.056z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders ok state 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fznZeY jliwHV"
      >
        <div
          class="sc-AxjAm hiFdOD"
        >
          <div
            class="sc-AxirZ bPICZX"
          >
            <select
              data-testid="select"
            >
              <option
                value=""
              >
                All
              </option>
              <option
                value="current"
              >
                Current
              </option>
              <option
                value="off_market"
              >
                Off Market
              </option>
              <option
                value="sold"
              >
                Sold
              </option>
              <option
                value="withdrawn"
              >
                Withdrawn
              </option>
            </select>
          </div>
        </div>
        <div
          class="sc-fzozJi ifZfUq"
        >
          <div
            class="sc-AxmLO fuihew"
          >
            <div
              class="sc-AxheI gTgauf"
            >
              <div
                class="sc-AxhUy fwYRBp"
              >
                <div
                  data-testid="price"
                >
                  $0
                </div>
                 
                <div
                  class="sc-AxiKw dPWazA"
                >
                  <span
                    data-testid="text"
                  />
                  <div
                    class="sc-AxhCb cXxvEK"
                    color="grey"
                    data-testid="dot"
                  />
                </div>
              </div>
              <img
                alt="undefined undefined"
                class="sc-Axmtr eepeAg"
                data-testid="image"
              />
              <div
                class="sc-AxgMl ftqCsy"
              >
                <br />
                <br />
                 
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fznZeY jliwHV"
    >
      <div
        class="sc-AxjAm hiFdOD"
      >
        <div
          class="sc-AxirZ bPICZX"
        >
          <select
            data-testid="select"
          >
            <option
              value=""
            >
              All
            </option>
            <option
              value="current"
            >
              Current
            </option>
            <option
              value="off_market"
            >
              Off Market
            </option>
            <option
              value="sold"
            >
              Sold
            </option>
            <option
              value="withdrawn"
            >
              Withdrawn
            </option>
          </select>
        </div>
      </div>
      <div
        class="sc-fzozJi ifZfUq"
      >
        <div
          class="sc-AxmLO fuihew"
        >
          <div
            class="sc-AxheI gTgauf"
          >
            <div
              class="sc-AxhUy fwYRBp"
            >
              <div
                data-testid="price"
              >
                $0
              </div>
               
              <div
                class="sc-AxiKw dPWazA"
              >
                <span
                  data-testid="text"
                />
                <div
                  class="sc-AxhCb cXxvEK"
                  color="grey"
                  data-testid="dot"
                />
              </div>
            </div>
            <img
              alt="undefined undefined"
              class="sc-Axmtr eepeAg"
              data-testid="image"
            />
            <div
              class="sc-AxgMl ftqCsy"
            >
              <br />
              <br />
               
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
